source/common/config/unified_mux/subscription_state.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/service/discovery/v3/discovery.pb.h"
+
+#include "source/common/config/ttl.h"
+#include "source/common/config/update_ack.h"
+#include "source/common/config/utility.h"
+#include "source/common/protobuf/protobuf.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Config {
+namespace UnifiedMux {
+
+// Tracks the protocol state of an individual ongoing xDS-over-gRPC session, for a single type_url.
+// There can be multiple SubscriptionStates active, one per type_url. They will all be
+// blissfully unaware of each other's existence, even when their messages are being multiplexed
+// together by ADS.
+// This is the abstract parent class for both the delta and state-of-the-world xDS variants.
+template <class RS, class RQ>
+class SubscriptionState : public Logger::Loggable<Logger::Id::config> {
+public:
+  // Note that, outside of tests, we expect callbacks to always be a WatchMap.
+  SubscriptionState(std::string type_url, UntypedConfigUpdateCallbacks& callbacks,
```

Comment:
Nit: `const std::string&` or `absl::string_view`

Reply:
please see the comment below.